connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -158,7 +158,48 @@ export async function getSiteNodesToSync(
   return nodeResources.map((r) => r.internalId);
 }
 
+export async function groupRootItemsByDriveId(
```

Comment:
afterthought: I think we can do all this without creating an activity, by doing `_.groupBy(nodeIds, getDriveIdFromInternalId)` (lodash helper) which would avoid calling the API on the items, wdyt?

Requires implementing `getDriveIdFromInternalId` that would get the drive id from the api path would be `internalIdFromTypeAndPath({ nodeType: "drive", itemAPIPath: {extract the first part `/drives/....` using typeAndPathFromInternalId(nodeId)` here

Not sure i'm clear?